rust/worker/src/blockstore/arrow/block/types.rs

```diff
@@ -206,10 +216,55 @@ impl Block {
     /// Returns the size of the block in bytes
     pub(crate) fn get_size(&self) -> usize {
         let mut total_size = 0;
+        let mut mock_size = 0;
+        let mut column_index = 0;
         for column in self.data.columns() {
+            let column_size = column.get_buffer_memory_size();
```

Comment:
this breaks if the underlying buffers are shared, which is true for if we load from Arrow IPC, so we see a size that is way larger than what it actually is.